app/api/units_api.rb

```diff
@@ -132,6 +134,14 @@ class UnitsApi < Grape::API
       end
     end
 
+    if unit_parameters[:portfolio_auto_generation_date]
```

Comment:
Move this to an active record validation - https://guides.rubyonrails.org/active_record_validations.html

See examples in Project and Unit. eg

```rb
# Unit
validates :code, uniqueness: { scope: :teaching_period, message: "%{value} already exists in this teaching period" }, if: :has_teaching_period?
# Project
validate :tutorial_enrolment_same_campus, if: :will_save_change_to_enrolled?
```

You can then remove these checks and they will trigger on Update! and Create!